inc/macro_utils/macro_utils.h

```diff
@@ -217,6 +217,12 @@ const char* MU_C3(MU_, enumIdentifier,_ToString)(enumIdentifier value)
 #define MU_TRUE_STRING "true"
 #define MU_BOOL_VALUE(v) MU_FALSE_STRING "\0" MU_TRUE_STRING + ((!!(v))*sizeof(MU_FALSE_STRING)), (v)
 
+/*PRI_TIME_T is to be used with a time_t variable - it produces strings such as "Tue May  4 14:42:17 2021"*/
+#define PRI_TIME_T "*.*s"
+
+/*TIME_T_VALUE is the counterpar of PRI_TIME_T in a printf/LogError instruction*/
+#define TIME_T_VALUE(t) (int)strlen(ctime(&t))-1, (int)strlen(ctime(&t))-1, ctime(&t)
```

Comment:
>ctime [](start = 36, length = 5)

Wondering if we could somehow avoid calling ctime that many times. And while wondering that I was wondering if the compiler is smart enough to avoid it :-).

Reply:
It is a good question. The reason why ctime is called is to chop out the \n that is mandated by "7.27.3.1 The asctime function" [C11].

The "standard" has a stock implementation to which any implementation of asctime should conform. Says "using the equivalent of the following algorithm". The implementation in the standard has a static char[26] array.

The leap of faith is to document these 26 characters (our of which we're only using 24) and write simply "24" there. Possibly I could add some tests too - just to make sure that at the time when the test is run, there's 25 characters returned. Why not? 

There are unclear 2 things for me: is this influenced by locale? Or does it always print English names? Other locales - if used - are they still good for the limit of 26 characters or do they used a different number of characters?

However - for the moment I will eliminate the strlen/ctime and use a hard coded value. And if our strings will start appearing ugly on the screen - well - we'll deal with that problem at that time.

Best Regards,
Andrei Porumb